### ISSUE TITLE
Login no longer working starting 2024

### ISSUE THREAD

**Issue by alXDraghiciu** [pnp/action-cli-login#35]
Hi, I was using the action to deploy an SPFX solution to the App Catalog.
Recently, the login step failed, my action looks like this:

```
      - name: Office 365 CLI Login
        uses: pnp/action-cli-login@v2.2.2
        with:
          TENANT: ${{ env.OFFICE365CLI_TENANT }}
          APP_ID: ${{ env.OFFICE365CLI_AADAPPID }}
          CERTIFICATE_ENCODED: ${{ secrets.CERTIFICATE_B64 }}
          CERTIFICATE_PASSWORD: ${{ secrets.CERTIFICATE_PASSWORD }}
```

I am getting the following error:

```
Run pnp/action-cli-login@v2.2.2
ℹ️ Installing CLI for Microsoft 365...
✅ CLI for Microsoft 365 successfully installed at /opt/hostedtoolcache/node/14.21.3/x64/bin/m365
ℹ️ Attempting to log in...
/opt/hostedtoolcache/node/14.21.3/x64/bin/m365 login --authType certificate --certificateBase64Encoded *** --password *** --appId *** --tenant ***
file:///opt/hostedtoolcache/node/14.21.3/x64/lib/node_modules/@pnp/cli-microsoft365/node_modules/ora/node_modules/string-width/index.js:27
	for (const {segment: character} of new Intl.Segmenter().segment(string)) {
	                                   ^
TypeError: Intl.Segmenter is not a constructor
    at stringWidth (file:///opt/hostedtoolcache/node/14.21.3/x64/lib/node_modules/@pnp/cli-microsoft365/node_modules/ora/node_modules/string-width/index.js:27:37)
    at Ora.#updateLineCount (file:///opt/hostedtoolcache/node/14.21.3/x64/lib/node_modules/@pnp/cli-microsoft365/node_modules/ora/index.js:197:45)
    at Ora.set text [as text] (file:///opt/hostedtoolcache/node/14.21.3/x64/lib/node_modules/@pnp/cli-microsoft365/node_modules/ora/index.js:140:24)
    at new Ora (file:///opt/hostedtoolcache/node/14.21.3/x64/lib/node_modules/@pnp/cli-microsoft365/node_modules/ora/index.js:59:[13](https://github.com/PC33Internal/wiki-parts/actions/runs/7694615804/job/20965745987#step:11:14))
    at ora (file:///opt/hostedtoolcache/node/[14](https://github.com/PC33Internal/wiki-parts/actions/runs/7694615804/job/20965745987#step:11:15).21.3/x64/lib/node_modules/@pnp/cli-microsoft365/node_modules/ora/index.js:377:9)
    at file:///opt/hostedtoolcache/node/14.21.3/x64/lib/node_modules/@pnp/cli-microsoft365/dist/cli/cli.js:25:[15](https://github.com/PC33Internal/wiki-parts/actions/runs/7694615804/job/20965745987#step:11:16)
    at ModuleJob.run (internal/modules/esm/module_job.js:183:25)
    at async Loader.import (internal/modules/esm/loader.js:[17](https://github.com/PC33Internal/wiki-parts/actions/runs/7694615804/job/20965745987#step:11:18)8:24)
    at async Object.loadESM (internal/process/esm_loader.js:68:5)
    at async handleMainPromise (internal/modules/run_main.js:59:12)
Error: 🚨 The process '/opt/hostedtoolcache/node/14.[21](https://github.com/PC33Internal/wiki-parts/actions/runs/7694615804/job/20965745987#step:11:22).3/x64/bin/m365' failed with exit code 1
Error: Error: The process '/opt/hostedtoolcache/node/14.21.3/x64/bin/m365' failed with exit code 1
```

Any idea how to fix this? Is this caused by GitHub deprecating NodeJS 14 actions?

**Comment by milanholemans** [comment]
Hi @alXDraghiciu that's possible indeed. We always encourage you to use the LTS version of Node. Have you tried `v20` yet?

**Comment by alXDraghiciu** [comment]
Thank you for your reply @milanholemans. Back when I started work on the SPFX project, Node 14 was recommended. [I see now they also support up to Node 18 LTS](https://learn.microsoft.com/en-us/sharepoint/dev/spfx/set-up-your-development-environment#install-nodejs).

I've updated the workflow file to use version `18.17.1` of Node JS and now everything works flawlessly with the action.

Thanks for the help!
